core/src/main/scala/org/apache/spark/executor/MesosExecutorBackend.scala

```diff
@@ -81,6 +90,14 @@ private[spark] class MesosExecutorBackend
       env)
   }
 
+  private def getRangesFromResources(executorInfo: ExecutorInfo): List[(Long, Long)] = {
+  executorInfo.getResourcesList.asScala.
+    filter(r => r.getType == Value.Type.RANGES & r.getName == "ports")
+    .toList.map(r => r.getRanges.getRangeList)
+    .flatMap { ranges => ranges.asScala.map { r => (r.getBegin, r.getEnd) }}
```

Comment:
This is a matter of debate, but check [monadic chaining](https://github.com/databricks/scala-style-guide#chaining) in the style guide. Might be good to break it into smaller chains.